lab_2_retrieval_w_bm25/main.py

```diff
@@ -48,6 +75,22 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
     In case of corrupt input arguments, None is returned.
     """
 
+    if not documents or not isinstance(documents, list) \
+            or not all(isinstance(lst_tokens, list) for lst_tokens in documents) \
+            or not all(lst_tokens for lst_tokens in documents):
+        return None
+
+    unique_words = []
+    for lst_tokens in documents:
+        for token in lst_tokens:
+            if not token or not isinstance(token, str):
+                return None
+            if token not in unique_words:
+                unique_words += [token]
+            else:
+                continue
```

Comment:
why do we need this?

Reply:
to create a list of unique words, without repetitions

Reply:
what about set?